### PR TITLE
multiclass experiments

### PR DESCRIPTION
**Pull request by BoltzmannBrain**

### CODE REVIEW

File: fluent/experiments/baseline_experiment.py [hunk 1]
```diff
@@ -39,7 +41,10 @@
 - classification and label are used interchangeably
 """
 
+## TODO: update docstrings
```
Comment (scottpurdy) [inline fluent/experiments/baseline_experiment.py]: Put this in the doc string

File: fluent/experiments/baseline_experiment.py [hunk 2]
```diff
@@ -51,18 +56,18 @@
 from fluent.utils.text_preprocess import TextPreprocess
 
 
-def runExperiment(model, patterns, labels, idxSplits):
+def runExperiment(model, patterns, idxSplits):
   """
```
Comment (scottpurdy) [inline fluent/experiments/baseline_experiment.py]: You should include a description of what this experiment does

File: fluent/experiments/baseline_experiment.py [hunk 3]
```diff
@@ -70,23 +75,24 @@ def runExperiment(model, patterns, labels, idxSplits):
 def training(model, trainSet):
   """Trains model on the bitmap patterns and corresponding labels lists."""
   for x in trainSet:
-    model.trainModel(x[0], x[1])
+    model.trainModel(x["pattern"], x["labels"])
```
Comment (scottpurdy) [inline fluent/experiments/baseline_experiment.py]: Is `x` a bitmap pattern? A sequence of bitmap patterns? Explain the parameters in the doc string and change this variable name to be more explicit.

File: fluent/experiments/baseline_experiment.py [hunk 4]
```diff
@@ -115,6 +121,39 @@ def computeExpectedAccuracy(predictedLabels, dataPath):
...
+  @param args       (Namespace)     User-provided arguments via the cmd line.
+  @param dataPath   (str)           Path where data is located.
+  @return           (tuple)         Tuple where first entry is a list of the
+      samples, the second is the list of gold labels per example, the third is
+      the list of all possible labels, and the fourth is the labels per example
+      in the data.
+  """
+  dataDict = readCSV(dataPath, 2, args.numClasses)
+
+  labelReference = list(set(
```
Comment (scottpurdy) [inline fluent/experiments/baseline_experiment.py]: Put a description for this. Something like "Combine all the values from dataDict into a single list and remove duplicates." Which is what I assume the intent is here.

File: fluent/experiments/baseline_experiment.py [hunk 5]
```diff
@@ -115,6 +121,39 @@ def computeExpectedAccuracy(predictedLabels, dataPath):
...
+      itertools.chain.from_iterable(dataDict.values())))
+
+  for k, v in dataDict.iteritems():
```
Comment (scottpurdy) [inline fluent/experiments/baseline_experiment.py]: `v` -> `labels` and `k` -> ???

File: fluent/experiments/experiment.py [hunk 6]
```diff
@@ -22,12 +22,15 @@
 Experiment runner for classification survey question responses.
 """
 
+## TODO: after merging PlotNLP code, set the defualt plotting to 1.
```
Comment (scottpurdy) [inline fluent/experiments/experiment.py]: Put in the doc string

File: fluent/experiments/runner.py [hunk 7]
```diff
@@ -72,49 +83,68 @@ def __init__(self,
...
+
+    self._preprocess(preprocess)
 
     if self.verbosity > 1:
-      for i, s in enumerate(self.samples):
-        print i, s, self.labelRefs[self.labels[i]]
+      for i, s in enumerate(self.samples): print i, s
 
 
+  ## TODO: does model need to know if multiclass??
```
Comment (scottpurdy) [inline fluent/experiments/runner.py]: What is this for? The function? Put in doc string. If something else then find a better palce for it

**Comment by scottpurdy** [comment]
Why does this have a bunch of @mihail911 and @BoltzmannBrain commits in it? You should be sending code in smaller chunks and you need a description in the PR. The commits should be just as descriptive, if not more than the PR since those are the official log. Avoid commits that aren't meaningful in the log.

**Comment by BoltzmannBrain** [comment]
Thank you for the review @scottpurdy. The changes will be reflected in #95, along with others in line with validating the multiclass experiments.

**Pull request merged**